datastream-server/src/main/java/com/linkedin/datastream/server/assignment/StickyPartitionAssignmentStrategy.java

```diff
@@ -592,4 +639,39 @@ private int getNumTasksForDatastreamFromZK(String stream) {
 
     return Integer.parseInt(_zkClient.readData(numTasksPath));
   }
+
+  /**
+   * Register metrics related to elastic task assignment
+   */
+  private void registerElasticTaskAssignmentMetrics(String taskPrefix) {
```

Comment:
Since the gauge metrics is registered per datastream, don't we need a deregister path when there are no tasks associated with the datastream prefix on this host.

Reply:
MetricsAware doesn't expose a deregister path. EventProducer, LiKafkaTransportProvider, etc fall into this category. Do you propose adding one? Connectors use a common metrics class which exposes de-registration code paths.

Reply:
As discussed offline, decided to go with ensuring the metric has the right value rather than deregistering it.